src/NovelRT/DotNet/RuntimeService.cpp

```diff
@@ -164,12 +164,12 @@ namespace NovelRT::DotNet
         tearDown();
     }
 
-    void RuntimeService::initialise()
+    void RuntimeService::initialise() noexcept
     {
         _exports.getActual().Initialise();
```

Comment:
I have a suspicion this can throw. Did you confirm its guaranteed not to?

Reply:
After looking at it a lot more I was finally able to understand what is happening here. Originally I thought that the constructor was assigning _exports with a set of pointers to functions, and throwing an exception if it couldnt find them. I just assumed that it was assigning some values to an Exports variable, oops.

What is actually happening is that the constructor is assigning _exports with a lambda function stored in the Lazy container. That lambda function returns an Exports struct with a set of function pointers.

So when RuntimeService::initialise() is called, _exports.getActual() takes the lambda assigned to the container, calls it, it returns an Exports, assigns them to _actual, and returns that set of function pointers.

THATS the step that can throw the exception, because thats when the lambda is actually called, and it has a throw in it.

Trying to see what .Initialise() is doing is hurting my brain, so I'll leave it at that for now!

I definitely messed up, but now I'm curious if the RuntimeService constructor can actually throw. Now that I looked at it for so long, its just assigning a bunch of lambda function variables. I have a feeling that it actually can't throw.

The whole class is super cool though.

Reply:
`noexcept(false)` is the exact same as not having `noexcept` at all. We prefer not having `noexcept` at all in this case.